betting-game-react/src/auth/Has/Has.tsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import jwt_decode from "jwt-decode";
+import { AuthorizationManager } from "../../auth";
+
+interface Props {
+    permissions: string[];
+    Yes?: React.ComponentType;
+    No?: React.ComponentType;
+}
+
+const Has: React.FC<Props> = ({ permissions, Yes, No }) => {
+    const [hasAccess, setHasAccess] = React.useState(false);
+    const AccessGranted: React.ComponentType =
+        Yes !== undefined ? Yes : () => null;
+    const AccessDenied: React.ComponentType =
+        No !== undefined ? No : () => null;
```

Comment:
wtf? do we really need to use Yes and No here instead of true/false?

Reply:
yep, the variables are called Yes and No, and depending on whether access is granted or not the component Yes or No is chosen. Usually, Yes is the component the user is trying to visit and No an error page.

Reply:
lol ok